public/app/features/correlations/utils.ts

```diff
@@ -54,18 +54,32 @@ const decorateDataFrameWithInternalDataLinks = (dataFrame: DataFrame, correlatio
     field.config.links = field.config.links?.filter((link) => link.origin !== DataLinkConfigOrigin.Correlations) || [];
     correlations.map((correlation) => {
       if (correlation.config?.field === field.name) {
-        const targetQuery = correlation.config?.target || {};
-        field.config.links!.push({
-          internal: {
-            query: { ...targetQuery, datasource: { uid: correlation.target.uid } },
-            datasourceUid: correlation.target.uid,
-            datasourceName: correlation.target.name,
-            transformations: correlation.config?.transformations,
-          },
-          url: '',
-          title: correlation.label || correlation.target.name,
-          origin: DataLinkConfigOrigin.Correlations,
-        });
+        if (correlation.type === CORR_TYPES.query.value && correlation.target !== undefined) {
+          const targetQuery = correlation.config?.target || {};
+          field.config.links!.push({
+            internal: {
+              query: { ...targetQuery, datasource: { uid: correlation.target.uid } },
+              datasourceUid: correlation.target.uid,
+              datasourceName: correlation.target.name,
+            },
+            url: '',
+            title: correlation.label || correlation.target.name,
+            origin: DataLinkConfigOrigin.Correlations,
+            meta: {
+              transformations: correlation.config?.transformations,
+            },
+          });
+        } else if (correlation.type === CORR_TYPES.external.value) {
+          const externalTarget = correlation.config.target;
+          if ('url' in externalTarget && externalTarget.url !== undefined) {
```

Comment:
The discriminate union should help with this kind of checks. TS would know `correlation.type === CORR_TYPES.external.value` because of this that here the type of `correlation` is `ExternalCorrelation` and so no additional checks would be needed for the `url` property